tests/test_browsing_context.py

```diff
@@ -149,32 +149,34 @@ async def test_browsingContext_getTreeWithNestedCrossOriginContexts_contextsRetu
 @pytest.mark.asyncio
 async def test_browsingContext_create_eventContextCreatedEmitted(
       websocket, context_id):
-    await subscribe(websocket, ["browsingContext.contextCreated"])
+    await subscribe(websocket, [
+        "browsingContext.contextCreated",
+        "browsingContext.domContentLoaded",
+        "browsingContext.load"])
 
     await send_JSON_command(websocket, {
         "id": 9,
         "method": "browsingContext.create",
         "params": {"type": "tab"}})
 
-    resp = await read_JSON_message(websocket)
-
-    if "params" in resp:
-        event = resp
-        command_result = await read_JSON_message(websocket)
-    else:
-        event = await read_JSON_message(websocket)
-        command_result = resp
-
-    new_context_id = event['params']['context']
-
-    # Assert "browsingContext.contextCreated" event emitted.
-    assert event == {
-        "method": "browsingContext.contextCreated",
-        "params": {
-            'context': new_context_id,
-            'parent': None,
-            'children': None,
-            'url': 'about:blank'}}
+    # Read 4 messages. The order can vary in headless and headful modes, so sort
+    # is needed:
+    # * `browsingContext.create` result.
+    # * `browsingContext.contextCreated` event.
+    # * `browsingContext.domContentLoaded` event.
+    # * `browsingContext.load` event.
+    messages = [await read_JSON_message(websocket),
+                await read_JSON_message(websocket),
+                await read_JSON_message(websocket),
+                await read_JSON_message(websocket)]
+
+    messages.sort(key=lambda x: x["method"] if "method" in x else "")
+    command_result = messages[0]
+    context_created_event = messages[1]
+    dom_content_loaded_event = messages[2]
+    load_event = messages[3]
```

Comment:
```suggestion
[command_result, context_created_event, dom_content_loaded_event, load_event] = messages
```

Reply:
If there are more than 4 values then you can do

`[command_result, context_created_event, dom_content_loaded_event, load_event, *_] = messages`

Reply:
TIL. Thanks!